### PR TITLE
Add PCA module

### PR DESCRIPTION
**Pull request by Raphael-Gronauer**
PCA python module

### CODE REVIEW

File: bin/PCA.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+parser.add_argument('-F','--figureType',default='png',help='File type of the plots in the output (Default = png)')
+
+args = parser.parse_args()
+data_path = args.input
+output_path = args.output
+sep = args.seperator
+color = args.color
+colC = args.colourCol
+dim = args.dimensions
+Annotations = args.annotationColumns
```
Comment (grst) [inline bin/PCA.py]: Sorry for being nitpicky, but per convention, in Python, all variables are lower case, while only classes start with a lower case letter.

File: bin/PCA.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    principalComponents = pca.fit_transform(x)
+    principalDf = pd.DataFrame(data = principalComponents, columns = ['principal component 1', 'principal component 2'])
+    
+    if sample == None:
+        finalDf = pd.concat([principalDf, data[targetList]], axis = 1)
+        for i in targetList:
+            colorList = finalDf[f'{i}']
+            colorList = [str(i) for i in colorList]
+            fig = px.scatter(principalComponents, x = 0 , y = 1, color = colorList,labels={'0': 'PC 1', '1': 'PC 2'}, color_discrete_sequence=getattr(px.colors.qualitative, color))
+            fig.write_image(f'{output_path}{name}_{i}.{figureType}')
```
Comment (grst) [inline bin/PCA.py]: This line is so long that it gets cropped on github, which is a pain when reviewing. 

Consider using the [black formatter](https://github.com/psf/black) for your python code. You may actually enjoy it, and it ends every discussion about code style ;)

**Review by grst — COMMENTED** [review]
Just two nasty code style comments :stuck_out_tongue:

File: bin/PCA.py (hunk 1 above)
Comment (Raphael-Gronauer) [reply bin/PCA.py]: Corrected

File: bin/PCA.py (hunk 2 above)
Comment (Raphael-Gronauer) [reply bin/PCA.py]: Corrected

**Comment by riederd** [comment]
Hi Raphael,
I fixed now a view issues and added the module to the main.nf. However it is still not running successfully, because I'm not sure about the different parameters the python script needs as input and I couldn't figure it out from the help messages.

e.g. what is `colour_colum_pca` and `annotation_columns_counts` and `annotation_columns_samplesheet`?

You might want to take a further look into this.

**Comment by Raphael-Gronauer** [comment]
Hi
colour_colum_pca:  is the column name which is used to color-code the dots in the PCA (like group1 group2...) can be more than one comma separated..
annotation_columns_counts: is a optional parameter if there are coulmns in the count matrix which are not actual count data like for example extra columns wit gene names or additional gene-id columns.
annotation_columns_samplesheet: are all columns in the sample-sheet which are not used to color-code the dots in the PCA.
added it to main.nf and to the param file.

**Comment by riederd** [comment]
Looks good now!

You should also include a publishing directive (publishDir) in nextflow.config
See examples there.

**Review by riederd — COMMENTED** [review]
Looks good now.
@grst any comments?

**Comment by grst** [comment]
CI is failing due to https://github.com/mamba-org/mamba/issues/1429, we really should switch to containers I guess :/ 

No other issues from my side, merging!

**Review by grst — APPROVED** [review]

**Pull request merged**